### PR TITLE
Move Drawer to GestureDetector

### PR DESCRIPTION
**Pull request by abarth**
This fixes an issue in the stocks app in horizontal mode where you could both
scroll and drag the drawer at the same time.

### CODE REVIEW

**Comment by collinjackson** [comment]
lgtm

File: sky/packages/sky/lib/src/widgets/drawer.dart [hunk 1]
```diff
@@ -132,41 +130,21 @@ class Drawer extends StatefulComponent {
...
-  // TODO(mpcomplete): Figure out how to generalize these handlers on a
-  // "PannableThingy" interface.
-  EventDisposition handlePointerDown(_) {
-    _performance.stop();
-    return EventDisposition.processed;
- }
-
-  EventDisposition handlePointerMove(sky.PointerEvent event) {
+  void _handleDragUpdate(double delta) {
     if (_performance.isAnimating)
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/drawer.dart]: how can the performance be animating in this case?

File: sky/packages/sky/lib/src/widgets/drawer.dart [hunk 2]
```diff
@@ -132,41 +130,21 @@ class Drawer extends StatefulComponent {
...
-    }
-    return EventDisposition.ignored;
+      return;
+    _performance.progress += delta / _kWidth;
   }
 
-  EventDisposition handlePointerCancel(_) {
-    if (!_performance.isAnimating) {
+  void _handleDragEnd(Offset velocity) {
+    if (_performance.isAnimating)
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/drawer.dart]: ditto

File: sky/packages/sky/lib/src/widgets/drawer.dart [hunk 3]
```diff
@@ -132,41 +130,21 @@ class Drawer extends StatefulComponent {
...
+      return;
+    if (velocity == Offset.zero) {
       _settle();
-      return EventDisposition.processed;
+    } else {
+      if (velocity.dx.abs() >= _kMinFlingVelocity)
+        _performance.fling(velocity: velocity.dx * _kFlingVelocityScale);
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/drawer.dart]: this doesn't seem right. Surely when you release a horizontal drag on a drawer, it should always either open or close.

File: sky/packages/sky/lib/src/widgets/drawer.dart [hunk 4]
```diff
@@ -132,41 +130,17 @@ class Drawer extends StatefulComponent {
...
+  void _handleDragUpdate(double delta) {
+    _performance.progress += delta / _kWidth;
   }
 
-  EventDisposition handlePointerUp(_) {
-    if (!_performance.isAnimating) {
+  void _handleDragEnd(Offset velocity) {
+    if (velocity.dx.abs() >= _kMinFlingVelocity) {
+      _performance.fling(velocity: velocity.dx * _kFlingVelocityScale);
+      return;
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/drawer.dart]: drop the return or the else

**Comment by abarth** [comment]
Done.

**Pull request merged**